apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java

```diff
@@ -313,61 +315,29 @@ public ApikeyController(ApikeyRepo apikeyRepo) {
-     *          Addionally, the following fields are (optionally) available in the response header:
-     *          - "X-RateLimit-Remaining" access usage number since the previous reset
-     *          - "X-RateLimit-Reset"     the number of seconds until the access usage count is reset
-     *          - "Apikey-not-found"      containing the string "apikey-not-found" is added when the Apikey
-     *                                    is not found, to help telling this HTTP 404 apart from one returned
-     *                                    by the webserver for other reasons
      */
-    @RequestMapping(path = "/{id}/validate", method = RequestMethod.POST)
-    public ResponseEntity<Apikey> validate(@PathVariable("id") String id,
-                                           @RequestParam(value = "api", required = false) String api,
-                                           @RequestParam(value = "method", required = false) String method) {
-
-        LOG.debug("validate apikey: {}", id);
-        ApiName     apiName; //TODO usage not implemented yet
-        HttpHeaders headers  = new HttpHeaders();
-        DateTime    nowDtUtc = new DateTime(DateTimeZone.UTC);
-        Date        now      = nowDtUtc.toDate();
-
-        if (StringUtils.isEmpty(api)) {
-            LOG.debug("no value for parameter 'api' supplied");
+    @PostMapping(path = "/validate")
+    public ResponseEntity<Apikey> validate(HttpServletRequest httpServletRequest) {
+        // When no apikey was supplied return 400
+        String id = getApikey(httpServletRequest);
+        if (null == id) {
+            LOG.debug(APIKEYMISSING);
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
-        } else if (!EnumUtils.isValidEnum(ApiName.class, api.toUpperCase())) {
-                LOG.debug("illegal value for parameter 'api': {}", api);
-                return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
-            }
-
-        if (StringUtils.isEmpty(method)) {
-            LOG.debug("no value for parameter 'method' supplied");
-        } else if (method.equalsIgnoreCase("krakboem")){
-            try {
-                int oops = 0 / 0;
-            } catch (ArithmeticException e) {
-                LOG.error("Deliberate error thrown: ", e);
-            }
-        } else if (!method.equalsIgnoreCase(READ) && !method.equalsIgnoreCase(WRITE)) {
-                return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
 
+        LOG.debug("validate apikey: {}", id);
+
         // retrieve apikey & check if available
         Apikey apikey = this.apikeyRepo.findOne(id);
         if (null == apikey) {
-            LOG.debug(APIKEYNOTFOUND + " with value: " + id);
-            headers.add(APIKEYNOTFOUND, APIKEYNOTFOUND.toLowerCase());
-            return new ResponseEntity<>(headers, HttpStatus.NOT_FOUND);
+            LOG.debug(APIKEYNOTREGISTERED, id);
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
```

Comment:
the bad request response should also indicate the reason. (e.g. invalid api key)